packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -500,12 +520,79 @@ export class AppSyncModelVisitor<
     }, [] as CodeGenField[]);
   }
 
+  /**
+   * Check out if primary key field exists in the model and generate primary key info
+   * @param model type defined with model directives
+   * @param directives directives defined at the type level of the model above
+   */
+  protected ensurePrimaryKeyField(model: CodeGenModel, directives: CodeGenDirective[]) {
```

Comment:
This is a long method, can we refactor this into `ensurePrimaryKeyFieldV1` and `ensurePrimaryKeyFieldV2`?

Reply:
Refactored to separate methods